Devscord.DiscordFramework/Framework/Commands/Services/BotCommandsParsingService.cs

```diff
@@ -1,58 +1,133 @@
-        }
+        public BotCommandsParsingService(BotCommandsPropertyConversionService botCommandPropertyConversionService) => this._botCommandPropertyConversionService = botCommandPropertyConversionService;
 
         public IBotCommand ParseRequestToCommand(Type commandType, DiscordRequest request, BotCommandTemplate template)
         {
-            var result = GetFilledInstance(commandType, template, x => request.Arguments.FirstOrDefault(a => a.Name.ToLowerInvariant() == x.ToLowerInvariant())?.Value);
+
+            var result = this.GetFilledInstance(commandType, template, (key, isList) => GetValueByName(key, isList, request, template));
             return result;
         }
 
+        private static object GetValueByName(string key, bool isList, DiscordRequest request, BotCommandTemplate template)
+        {
+            var result = request.Arguments.FirstOrDefault(a => a.Name.ToLowerInvariant() == key.ToLowerInvariant());
+            if (result == null)
+            {
+                return null;
+            }
+
+            var argType = template.Properties.FirstOrDefault(x => x.Name.ToLowerInvariant() == key.ToLowerInvariant())?.Type;
+            if (argType.HasValue && argType.Value == BotCommandPropertyType.Bool)
+            {
+                return result.Value ?? bool.TrueString;
+            }
+
+            if (!isList)
+            {
+                return result.Value;
+            }
+
+            var argumentsList = request.Arguments.ToList();
+            var indexOf = argumentsList.IndexOf(result);
+            var nextResults = argumentsList.Skip(indexOf + 1).TakeWhile(x => x.Name == null);
+            var list = new List<string> { result.Value };
+            list.AddRange(nextResults.Select(x => x.Value));
+            return list;
+        }
+
         public IBotCommand ParseCustomTemplate(Type commandType, BotCommandTemplate template, Regex customTemplate, string input)
         {
             var match = customTemplate.Match(input);
-            if(!this.CustomTemplateIsValid(match, template))
+            if (!this.CustomTemplateIsValid(match, template))
             {
                 Log.Warning("Custom template {customTemplate} is not valid for {commandName}", customTemplate, template.CommandName);
                 return null;
             }
-            var result = GetFilledInstance(commandType, template, x => match.Groups.ContainsKey(x) ? match.Groups[x].Value : null);
+
+            var result = this.GetFilledInstance(commandType, template, (key, isList) => GetValueByNameFromCustomCommand(key, isList, template, match));
             return result;
         }
 
-        private IBotCommand GetFilledInstance(Type commandType, BotCommandTemplate template, Func<string, string> getValueByName)
+        private static object GetValueByNameFromCustomCommand(string key, bool isList, BotCommandTemplate template, Match match)
```

Comment:
tak w sumie to można to przenieść do osobnego serwisu
bo obecny sie troche rozrósł